app/scripts/lib/signature/util.ts

```diff
@@ -0,0 +1,27 @@
+import type {
+  OriginalRequest,
+  TypedMessageParams,
+} from '@metamask/message-manager';
+import { endTrace, TraceName } from '../../../../shared/lib/trace';
+
+export type SignatureParams = [TypedMessageParams, OriginalRequest];
+
+export type AddTypedMessageRequest = {
+  signatureParams: SignatureParams;
+  newUnsignedTypedMessage: (...args: SignatureParams) => Promise<string>;
```

Comment:
Rather than passing a callback, would it make sense to encapsulate the usage of the `SignatureController` in this file and so pass the controller instance in the request?